plc4j/tools/connection-cache/src/main/java/org/apache/plc4x/java/utils/cache/LeasedPlcConnection.java

```diff
@@ -80,6 +80,11 @@ public synchronized void close() {
         connectionContainer.returnConnection(this, invalidateConnection);
     }
 
+    @Override
+    public Optional<PlcTag> parseTagAddress(String tagAddress) {
+        return Optional.empty();
```

Comment:
I think this should probably forward to the driver being used and not simply return empty.

Reply:
> I think this should probably forward to the driver being used and not simply return empty.

Requesting assistance, as I am not very familiar with the overall code of the project, PlcTagHandler was not found within LeasedPlcConnection. However, LeasedPlcConnection is an implementation class of PlcConnection. Is it appropriate to directly use connection.get().parseTagAddress(tagAddress)?

Reply:
Leased connection has a "connection" property which refers to the actual connection ... so I guess something like this should work:

```
    @Override
    public Optional<PlcTag> parseTagAddress(String tagAddress) {
        PlcConnection plcConnection = connection.get();
        if(plcConnection == null) {
            throw new PlcRuntimeException("Error using leased connection after returning it to the cache.");
        }
        return plcConnection.parseTagAddress(tagAddress);
    }

```

Reply:
Already modified.